spec/system/viewing_sidebar_spec.rb

```diff
@@ -5,34 +5,62 @@
   fab!(:user) { Fabricate(:user) }
   fab!(:category_sidebar_section_link) { Fabricate(:category_sidebar_section_link, user: user) }
 
+  before_all { sign_in(user) }
+
   describe "when using the legacy navigation menu" do
     before { SiteSetting.navigation_menu = "legacy" }
 
-    it "should display the sidebar when `enable_sidebar` query param is '1'" do
-      sign_in(user)
-
-      visit("/latest?enable_sidebar=1")
+    it "should display the sidebar when `navigation_menu` query param is 'sidebar'" do
+      visit("/latest?navigation_menu=sidebar")
 
       sidebar = PageObjects::Components::Sidebar.new
 
       expect(sidebar).to be_visible
       expect(sidebar).to have_category_section_link(category_sidebar_section_link.linkable)
       expect(page).not_to have_css(".hamburger-dropdown")
     end
+
+    it "should display the sidebar dropdown menu when `navigation_menu` query param is 'header_dropdown'" do
+      visit("/latest?navigation_menu=header_dropdown")
+
+      sidebar = PageObjects::Components::Sidebar.new
+
+      expect(sidebar).to be_not_visible
```

Comment:
nit, this reads a bit weird, kind of like Shakespeare. I am pretty sure `not_to be_visible` works too?

Reply:
`not_to be_visible` works but it is an anti-pattern because "visible" is defined as "page.has_css?". If we do `not_to have_css` here, it'll end up making the tests slower since it waits for the element to appear. Therefore, I had to define a `not_visible?` method which relies on `page.has_no_css?`.